scripts/irods/upgrade_configuration.py

```diff
@@ -214,6 +275,39 @@ def update_base(base, updates):
 
     return new_server_config
 
+def convert_to_v5_schema_and_add_missing_properties(server_config):
+    def update_base(base, updates):
+        for k, v in updates.items():
+            if isinstance(v, collections.abc.Mapping):
+                base[k] = update_base(base.get(k, {}), v)
+            else:
+                base[k] = v
+        return base
+
+    # Load server_config.json.template as our base.
+    with open(paths.get_template_filepath(paths.server_config_path())) as f:
+        base = json.load(f)
+
+    new_server_config = update_base(base, server_config)
+
+    # Prompt the admin for the host which uniquely identifies this server, but only
+    # if the property doesn't exist in the new server_config.json configuration.
+    if 'host' not in new_server_config:
+        new_server_config['host'] = lib.default_prompt(
+            'iRODS server FQDN, hostname, or IP (256 characters max)',
+            default=lib.get_hostname(),
+            input_filter=lib.character_count_filter(minimum=1, maximum=256, field='iRODS server host'))
```

Comment:
Need to figure out how to pull the `irods_host` value from the service account's irods_environment.json file.

At this level, we don't have access to that information due to how upgrades are designed/implemented.

Reply:
I've updated this to simply load the local irods_environment.json file and pull the value of `irods_host` from it.

Will confirm it works by going through the upgrade process.